x-pack/platform/plugins/shared/streams/server/lib/streams/state_management/streams/wired_stream.ts

```diff
@@ -414,6 +412,20 @@ export class WiredStream extends StreamActiveRecord<Streams.WiredStream.Definiti
     return { isValid: true, errors: [] };
   }
 
+  private async getMatchingDataStream() {
```

Comment:
nit we could tweak this to return `{ existsAsIndex: boolean; existsAsManagedStream: boolean; }` and embed the `managed_by` logic to reuse it in both places